kategory-test/src/main/kotlin/kategory/laws/TraverseFilterLaws.kt

```diff
@@ -0,0 +1,25 @@
+package kategory
+
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+import kategory.Option.Some
+import kategory.Option.None
+
+object TraverseFilterLaws {
+
+    inline fun <reified F> laws(TF: TraverseFilter<F> = traverseFilter<F>(), GA: Applicative<F> = applicative<F>(), crossinline cf: (Int) -> HK<F, Int>, EQ: Eq<HK<F, Int>>, EQ_NESTED: Eq<HK<F, HK<F, Int>>> = Eq.any()): List<Law> =
+            TraverseLaws.laws(TF, GA, cf, EQ) + listOf(
+                    Law("TraverseFilter Laws: Identity", { identityTraverseFilter(TF, GA, cf, EQ_NESTED) }),
+                    Law("TraverseFilter Laws: filterA consistent with TraverseFilter", { filterAconsistentWithTraverseFilter(TF, GA, cf, EQ_NESTED) })
+            )
+
+    inline fun <reified F> identityTraverseFilter(FT: TraverseFilter<F>, GA: Applicative<F> = applicative<F>(), crossinline cf: (Int) -> HK<F, Int>, EQ: Eq<HK<F, HK<F, Int>>> = Eq.any()) =
+            forAll(genConstructor(genIntSmall(), cf), { fa: HK<F, Int> ->
```

Comment:
If you have an applicative of F you shouldn't need a constructor function. Applicative::pure does the same job!